lib/Parse/ParsedRawSyntaxRecorder.cpp

```diff
@@ -65,39 +79,42 @@ ParsedRawSyntaxRecorder::recordToken(tok tokKind, CharSourceRange tokRange,
   CharSourceRange range(offset, length);
   OpaqueSyntaxNode n =
       SPActions->recordToken(tokKind, leadingTrivia, trailingTrivia, range);
-  return ParsedRawSyntaxNode(n, range, SyntaxKind::Token, tokKind,
-                             ParsedRawSyntaxNode::DataKind::Recorded,
-                             /*IsMissing=*/false);
+  return makeParsedRawSyntaxNode(n, SyntaxKind::Token, tokKind,
+                                 ParsedRawSyntaxNode::DataKind::Recorded,
+                                 /*IsMissing=*/false, range);
 }
 
 ParsedRawSyntaxNode
 ParsedRawSyntaxRecorder::recordMissingToken(tok tokenKind, SourceLoc loc) {
-  CharSourceRange range{loc, 0};
   OpaqueSyntaxNode n = SPActions->recordMissingToken(tokenKind, loc);
-  return ParsedRawSyntaxNode(n, range, SyntaxKind::Token, tokenKind,
-                             ParsedRawSyntaxNode::DataKind::Recorded,
-                             /*isMissing=*/true);
+  return makeParsedRawSyntaxNode(n, SyntaxKind::Token, tokenKind,
+                                 ParsedRawSyntaxNode::DataKind::Recorded,
+                                 /*isMissing=*/true, CharSourceRange(loc, 0));
 }
 
 ParsedRawSyntaxNode
 ParsedRawSyntaxRecorder::recordRawSyntax(SyntaxKind kind,
                                          MutableArrayRef<ParsedRawSyntaxNode> elements) {
   assert(kind != SyntaxKind::Token && "Use recordToken to record a token");
-#ifndef NDEBUG
+#ifdef PARSEDRAWSYNTAXNODE_VERIFY_RANGES
   ParsedRawSyntaxRecorder::verifyElementRanges(elements);
 #endif
```

Comment:
Probably `verifyElementRanges()` should return the total range. So
```cpp
#ifdef PARSEDRAWSYNTAXNODE_VERIFY_RANGES
auto range = ParsedRawSyntaxRecorder::verifyElementRanges(elements);
#endif
...
return makeParsedRawSyntaxNode(..., range);
```
This way we only have one `#ifdef` in a function.

Reply:
Nice idea. I’ll try it. Would definitely make things easier.

Reply:
Or even
```
return makeParsedRawSyntaxNode(..., getLayoutRange(elements))
```
without any `#ifdef`